src/builder.rs

```diff
@@ -3,151 +3,244 @@ use opentelemetry::{
+    /// them available for querying using APL, the Axiom SDK or the Axiom CLI.
+    #[must_use]
     pub fn with_dataset(mut self, dataset_name: impl Into<String>) -> Self {
         self.dataset_name = Some(dataset_name.into());
         self
     }
 
     /// Set the Axiom API token to use.
+    #[must_use]
     pub fn with_token(mut self, token: impl Into<String>) -> Self {
         self.token = Some(token.into());
         self
     }
 
     /// Set the Axiom API URL to use. Defaults to Axiom Cloud.
+    #[must_use]
     pub fn with_url(mut self, url: impl Into<String>) -> Self {
         self.url = Some(url.into());
         self
     }
 
     /// Set the trace config.
+    #[must_use]
     pub fn with_trace_config(mut self, trace_config: impl Into<TraceConfig>) -> Self {
         self.trace_config = Some(trace_config.into());
         self
     }
 
     /// Set the service name. It will be set as a resource attribute with the
     /// name `service_name`.
+    #[must_use]
     pub fn with_service_name(mut self, service_name: impl Into<String>) -> Self {
         self.service_name = Some(service_name.into());
         self
     }
 
     /// Don't fall back to environment variables.
+    #[must_use]
     pub fn no_env(mut self) -> Self {
         self.no_env = true;
         self
     }
 
+    /// Set the resource tags for the open telemetry tracer that publishes to Axiom.
+    /// These tags will be added to all spans.
+    ///
+    /// # Errors
+    ///
+    /// Returns an error if a key is invalid.
+    #[must_use]
+    pub fn with_tags(mut self, tags: &[(&str, &str)]) -> Self {
+        self.tags = tags
+            .iter()
+            .map(|(k, v)| KeyValue::new(Key::from((*k).to_string()), (*v).to_string()))
+            .collect::<Vec<_>>();
+        self
+    }
+
     /// Initialize the global subscriber. This panics if the initialization was
     /// unsuccessful, likely because a global subscriber was already installed or
     /// `AXIOM_TOKEN` is not set or invalid.
-    pub fn init(self) -> Guard {
-        self.try_init().unwrap()
+    ///
+    /// # Errors
+    ///
+    /// Returns an error if the initialization was unsuccessful, likely because
+    /// a global subscriber was already installed or `AXIOM_TOKEN` is not set or
+    /// invalid.
+    ///
+    pub fn init(self) -> Result<(), Error> {
+        self.try_init()
     }
 
     /// Initialize the global subscriber. This returns an error if the
     /// initialization was unsuccessful, likely because a global subscriber was
     /// already installed or `AXIOM_TOKEN` is not set or invalid.
-    pub fn try_init(self) -> Result<Guard, Error> {
-        let (layer, guard) = self.layer()?;
+    ///
+    /// # Errors
+    ///
+    /// Returns an error if the initialization was unsuccessful, likely because
+    /// a global subscriber was already installed or `AXIOM_TOKEN` is not set or
+    /// invalid.
+    pub fn try_init(self) -> Result<(), Error> {
```

Comment:
Can we delete this and only have one `init`?

Reply:
Yes, good catch. I've no preference but `init` saves typing `try_` so i'll do that

Reply:
Consolidated on `init()`